crates/ruff_python_formatter/src/statement/stmt_if.rs

```diff
@@ -96,35 +48,56 @@ impl FormatNodeRule<StmtIf> for FormatStmtIf {
     }
 }
 
-fn else_if(or_else: &Suite) -> Option<IfOrElIf> {
-    if let [Stmt::If(if_stmt)] = or_else.as_slice() {
-        Some(IfOrElIf::ElIf(if_stmt))
-    } else {
-        None
-    }
-}
-
-enum IfOrElIf<'a> {
-    If(&'a StmtIf),
-    ElIf(&'a StmtIf),
-}
-
-impl<'a> IfOrElIf<'a> {
-    const fn statement(&self) -> &'a StmtIf {
-        match self {
-            IfOrElIf::If(statement) => statement,
-            IfOrElIf::ElIf(statement) => statement,
-        }
-    }
+/// Note that this implementation misses the leading newlines before the leading comments because
+/// it does not have access to the last node of the previous branch
```

Comment:
What does that mean? Is this a note for callers that they need to insert that newline manually?

Reply:
kinda? `FormatElifElseClause` is currently unused, i added it more for completeness, the real usage is through `format_elif_else_clause` which handles the trailing newline

Reply:
I think this is a case where it would even be fair to panic because it is a programmatic error that needs to be fixed

Reply:
in that case i just wouldn't implement `FormatNodeRule` for `ElifElseClause`, i mainly added it because it's effectively free and i can imagine use cases where we want to format a fixed elif branch and newline more or less is acceptable